File: cm_collectors_server/datatype/search.dataType.go
package datatype

const V_Search_Not string = "not"
const V_Search_Before2000 string = "before_2000"

type ReqParam_SearchData struct {
	SearchTextSlc []string                 `json:"searchTextSlc"`
	Sort          E_searchSort             `json:"sort"`
	Country       I_searchGroup            `json:"country"`
	Definition    I_searchGroup            `json:"definition"`
	Year          I_searchGroup            `json:"year"`
	Star          I_searchGroup            `json:"star"`
	Performer     I_searchGroup            `json:"performer"`
	Cup           I_searchGroup            `json:"cup"`
	Tag           map[string]I_searchGroup `json:"tag"`
}

type E_searchSort string

const (
	E_searchSort_addTimeDesc     E_searchSort = "addTimeDesc"
	E_searchSort_addTimeAsc      E_searchSort = "addTimeAsc"
	E_searchSort_issueNumberDesc E_searchSort = "issueNumberDesc"
	E_searchSort_issueNumberAsc  E_searchSort = "issueNumberAsc"
	E_searchSort_scoreDesc       E_searchSort = "scoreDesc"
	E_searchSort_scoreAsc        E_searchSort = "scoreAsc"
	E_searchSort_starDesc        E_searchSort = "starDesc"
	E_searchSort_starAsc         E_searchSort = "starAsc"
	E_searchSort_issuingDateDesc E_searchSort = "issuingDateDesc"
	E_searchSort_issuingDateAsc  E_searchSort = "issuingDateAsc"
	E_searchSort_titleDesc       E_searchSort = "titleDesc"
	E_searchSort_titleAsc        E_searchSort = "titleAsc"
	E_searchSort_history         E_searchSort = "history"
	E_searchSort_hot             E_searchSort = "hot"
	E_searchSort_youLike         E_searchSort = "youLike"
)

type I_searchGroup struct {
	Logic   E_searchLogic `json:"logic"`
	Options []string      `json:"options"`
}

type E_searchLogic string

const (
	E_searchLogic_single   E_searchLogic = "single"
	E_searchLogic_multiAnd E_searchLogic = "multiAnd"
	E_searchLogic_multiOr  E_searchLogic = "multiOr"
	E_searchLogic_not      E_searchLogic = "not"
)

File: cm_collectors_html/src/dataType/search.dataType.ts
export interface I_searchData {
  searchTextSlc: string[]; // 搜索文本
  sort: E_searchSort; // 排序
  country: I_searchGroup;  // 国家
  definition: I_searchGroup; // 清晰度
  year: I_searchGroup; // 年份
  star: I_searchGroup; // 评星
  performer: I_searchGroup; // 演员
  cup: I_searchGroup; // cup
  tag: Record<string, I_searchGroup>; // 标签
}

export interface I_searchGroup {
  logic: E_searchLogic;
  options: string[];
}

export enum E_searchLogic {
  Single = 'single',         // 单选
  MultiAnd = 'multiAnd',     // 多选（与）
  MultiOr = 'multiOr',       // 多选（或）
  Not = 'not'                // 非
}

export enum E_searchSort {
  AddTimeDesc = 'addTimeDesc', //时间倒叙
  AddTimeAsc = 'addTimeAsc', //时间正序
  IssuingDateDesc = 'issuingDateDesc', //发行时间倒叙
  IssuingDateAsc = 'issuingDateAsc', //发行时间正序
  IssueNumberDesc = 'issueNumberDesc', //发行倒叙
  IssueNumberAsc = 'issueNumberAsc',  //发行正序
  ScoreDesc = 'scoreDesc', //评分倒叙
  ScoreAsc = 'scoreAsc', //评分正序
  StarDesc = 'starDesc', //评星倒叙
  StarAsc = 'starAsc', //评星正序
  TitleDesc = 'titleDesc', //标题倒叙
  TitleAsc = 'titleAsc',  //标题正序
  History = 'history', //历史记录
  Hot = 'hot', //当前热度
  YouLike = 'youLike', //猜你喜欢
}
